### ISSUE TITLE
Cannot build Arcane Fortune

### ISSUE THREAD

**Issue by ScumbagDog** [cody2007/arcane_fortune#2]
Running latest commit 4b683be7be0dec95592e2e591ce045df4149a6ce
Cargo v1.46.0 x64

I'm assuming the way to build Arcane Fortune is to cd into arcane_fortune/arcane_fortune and run cargo build, but the ironnetpro dependency isn't being loaded correctly which throws the following error (~ in place of my home dir path):
```

Caused by:
  failed to load source for dependency `ironnetpro`

Caused by:
  Unable to update ~/ironnetpro

Caused by:
  failed to read `~/ironnetpro/Cargo.toml`
```

I think this is because the cargofile in Arcane Fortune refers to the ironnet path as "../../ironnetpro", rather than "../ironnetpro".
I tried doing this, but I still couldn't build the project, due to the versioning system referring to a nonexisting .git folder in ironnetpro.

What am I doing wrong?

**Comment by cody2007** [comment]
Thanks for letting me know. I'll fix these issues soon -- you're not doing anything wrong it's due to last minute file re-organizations I made.

Here's what's going wrong:

On my system I have the directory arrangements a bit different, and forgot to update the Cargo.toml when I pushed everything into this repository (seems that you fixed that part on your system).

The second issue is related to the first -- also on my system I have ironnetpro in a separate (non-public) git repository and have the build process auto-embed the git commit ID of the ironnetpro crate into the main AF binary. For the mean time until I push a fix, you might be able to simply init a git repository in the ironnetpro directory and then try again.

-------------------------------
Since I haven't yet written up any help file for building the game here's a few other things (some of them not ideal) that might be useful to know:

-I compile and run in the arcane_fortune/src directory. The game will attempt to load its config files via relative path and looks for the directory 'config/'. You also need to copy the neural network weight directory from the binary release because they're not saved in this repository due to space concerns. copy the 'game/nn' directory to arcane_fortune/src/nn'

cargo run --release # builds the ncurses version using your local ncurses install
cargo run --release --features sdl # builds the game with full mouse support

You may get errors about the linking of the SDL or ncurses libraries. I generally resolve that by looking for the link directory that cargo seems to be using based on its error message and copying the shared object files there.

--------------------------------
more minor notes:

-The statically compiled version for linux (using musl) for some reason crashes when run in debug mode. This didn't happen until I upgraded to rust 0.46.0
-I do not know about compiling the game on Windows and only cross-compile from Linux. I had issues a while ago, but don't remember what they were.

**Comment by cody2007** [comment]
*rust 1.46.0, not 0.46.0

**Comment by cody2007** [comment]
I think the issues you raised should be fixed now.

I also added "compiling.txt" in the root of the repository for more instructions, similar to what I already said here.

Let me know if you want more help or have issues!

**Comment by ghost** [comment]
Hi @cody2007, I'm also having issues compiling -- it seems the build requires some git info for ironnetpro:
```
arcane_fortune/src/disp/version_status.rs:13:const IRONNETPRO_COMMIT_ID: &str = include_str!("../../../../ironnetpro/.git/refs/heads/master");
```
but the ironnetpro git subtree is not included in the repo. You may need to split off the ironnetpro into a submodule if the .git tree needs to be present.

If I replace the file read with a static string, the build finishes okay (some warnings about unused vars, functions and imports, but nothing important).

**Comment by cody2007** [comment]
Oh, sorry looks like I only fixed half the issue.
I pushed another small change. Try again with the latest commit on the v0.2.0 branch.

There are compiler warnings that I'm aware about but shouldn't pose problems for you.

(Here's context that you don't necessarily need to know: Ironnetpro is a limited but somewhat general purpose machine learning library for GPU or CPU neural network training and inference. But I decided to put it into the game's repository directly to avoid version incompatibilities.

If I update ironnetpro in a way that breaks the game, I'll also update the game. So anyone that wants to build the game can just checkout any commit or branch of the game and know that they're getting the correct version of ironnetpro. Maybe there's a better way to do this but this is the simplest I came up with for now.)
